R/print.data.table.R

```diff
@@ -164,3 +185,29 @@ shouldPrint = function(x) {
 #   as opposed to printing a blank line, for excluding col.names per PR #1483
 cut_top = function(x) cat(capture.output(x)[-1L], sep = '\n')
 
+# to calculate widths of data.table and console
+nchar_width = function(x) {
+  max(nchar(as.character(x), type = "width"))
+}
+dt_width = function(x, class, row.names, names) {
+  # gets the width of the data.table at each column
+  #   and compares it to the console width
+  widths = sapply(x, nchar_width)
+  if (class) widths = ifelse(widths < 6, 6, widths)
+  names = sapply(names, nchar_width)
+  dt_widths = ifelse(widths > names, widths, names)
+  rownum_width = if (row.names) nchar_width(nrow(x)) else 0
+  cumsum(dt_widths + 1) + rownum_width + 2
```

Comment:
Yes, you are correct. My personal preference is the way I've done it but that, by no means, is a good reason. What do you suggest? Would having another option be the fix here?

Reply:
I think we should pick one and stick with it rather than add more options. We can play around with it in dev for a while & maybe change later.

Just some more thoughts: I guess most common is to have the first few columns with keys/"ID"-style rows. Then have "data" in the "rightmost" columns. So the tradeoff is -- showing all the key columns, vs showing some keys, some data. Still don't have a feel yet for which is more helpful to see in a preview.

Reply:
That makes sense. I like the idea of differentiating between the keys and "data" in the preview. Definitely could get to some weird edge cases with this, depending on the console width and the number of keys...